### PR TITLE
Ensure network manager is installed

### PR DESCRIPTION
**Pull request by technowhizz**
Adds a task to the network manager playbook to ensure that the network-manager package is installed on Ubuntu.

### CODE REVIEW

File: tasks/networkmanager.yml [hunk 1]
```diff
@@ -1,4 +1,11 @@
 ---
+- name: Ensure NetworkManager is installed (Ubuntu)
+  ansible.builtin.package:
+    name: network-manager
+    state: present
+  become: true
+  when: ansible_facts['distribution'] == "Ubuntu"
```
Comment (Alex-Welsh) [inline tasks/networkmanager.yml:7]: You could remove this condition so it always checks, which would also catch the same issue in other distributions

Comment (markgoddard) [reply tasks/networkmanager.yml:7]: Is this for a Kayobe/OpenStack environment? We use systemd-networkd on Ubuntu

Comment (technowhizz) [reply tasks/networkmanager.yml:7]: Yes it is. However without network manager installed, this seems to fail ``name: Ensure the `VXLAN` configuration is present within `/etc/NetworkManager/system-connections``

Comment (mnasiadka) [reply tasks/networkmanager.yml:7]: So we should rather fix the failing task, not install NetworkManager on Ubuntu?

Comment (mnasiadka) [reply tasks/networkmanager.yml:7]: Scratch that, it seems that role does not support Ubuntu at all.

Comment (technowhizz) [reply tasks/networkmanager.yml:7]: The whole stackhpc.vxlan role?

Comment (mnasiadka) [reply tasks/networkmanager.yml:7]: Yes, it only supports Network Scripts (/etc/sysconfig/network-scripts/) or Network Manager - and on Ubuntu we use systemd-networkd.

Comment (technowhizz) [reply tasks/networkmanager.yml:7]: So would we create a new role for systemd-networkd?

Comment (technowhizz) [reply tasks/networkmanager.yml:7]: Nvm will just add a systemd-netword.yml file

Comment (markgoddard) [reply tasks/networkmanager.yml:7]: Can you give some background on what you're trying to do? Run the multinode env on Ubuntu? Currently it's not supported, mainly due to this role. IIRC we had a brief look at vxlan in systemd-networkd, but it appeared to not be fully supported in ubuntu focal. But that might be wrong/outdated.

Comment (technowhizz) [reply tasks/networkmanager.yml:7]: @markgoddard Yes that is the plan. I'm now investigating trying to add vxlan to systemd-networkd. I'm testing it on Jammy, hopefully I have some luck

**Comment by technowhizz** [comment]
Need to work on adding vxlan support to systemd-networkd instead of band aiding with network manager as @mnasiadka and @markgoddard pointed out

**Pull request closed without merging**